rustls/src/client/hs.rs

```diff
@@ -455,11 +455,13 @@ impl ExpectServerHello {
 
     fn into_expect_tls12_new_ticket_resume(
         self,
+        negotiated_version: ProtocolVersion,
```

Comment:
By this point we're absolutely certain we're talking TLS1.2, therefore there's no need to keep that information separately here, or below.